ement-room.el

```diff
@@ -2687,18 +2681,13 @@ Format defaults to `ement-room-message-format-spec', which see."
             (insert-and-inherit
              (propertize " "
                          'display `((margin right-margin) ,string))))))
-      (when-let* ((data (map-elt (ement-event-local event) (intern (concat "url-previews-"
-                                                                           (ement-event-id event)))))
-                  (num (or 1 (map-elt data
-                                      (intern (concat "url-previews-" (ement-event-id event)))))))
+      (when-let* ((data (ement-event-local event))
+                  (num (map-elt data 'url-preview-n)))
         (goto-char (point-max))
-        (unless (looking-at-p "^\n")
-          (insert "\n"))
-        (cl-loop repeat num
-                 do (when-let ((inner-data (map-elt data (intern (concat "preview-"
-                                                                         (number-to-string num))))))
-                      (insert (ement-room--format-url-preview inner-data))))
-        )
+        (cl-loop for i from 1 to num
```

Comment:
...rather than iterating through them like this, you just `(dolist (preview previews) ...)`.